src/services/repairRequest.ts

```diff
@@ -1,26 +1,45 @@
+import { RepairRequestImage } from "@prisma/client";
+
+import { presignImage, presignImages } from "@/lib/presign-image";
 import userService from "@/services/user";
 import { RepairRequest, RepairRequestResponse } from "@/types";
 
+/**
+ * Converts a list of RepairRequest to RepairRequestResponse
+ * @param repairRequests
+ * @param withImage If you need the image presigned urls
+ * @returns
+ */
 const toClientResponse = async (
-  repairRequests: RepairRequest[]
+  repairRequests: Array<RepairRequest & { images?: RepairRequestImage[] }>,
+  withImage = false
 ): Promise<RepairRequestResponse[]> => {
   const userIds = repairRequests.flatMap((e) => [e.createdBy, e.assignedTo]);
-
   const userMap = await userService.getUserMapFromIds(userIds);
 
-  const responses: RepairRequestResponse[] = repairRequests.map((req) => {
-    return {
-      ...req,
-      createdBy:
-        userMap[req.createdBy] ?? userService.unknownUser(req.createdBy),
-      assignedTo:
-        userMap[req.assignedTo] ?? userService.unknownUser(req.assignedTo),
-      requestDate: req.requestDate.toISOString(),
-      updatedAt: req.updatedAt.toISOString(),
-      hoursWorked: req.hoursWorked.toNumber(),
-      thumbnailImage: req.thumbnailImage ?? undefined
-    };
-  });
+  const responses: RepairRequestResponse[] = await Promise.all(
+    repairRequests.map(async (req) => {
+      const [thumbnailImage, images] = withImage
+        ? await Promise.all([
+            presignImage(req.thumbnailImage),
+            presignImages(req.images?.map(({ s3Key }) => s3Key) || [])
+          ])
```

Comment:
This does not work currently as req.images will always return undefined. You will need to include the images relation for the repair request query. https://www.prisma.io/docs/orm/prisma-client/queries/relation-queries

Reply:
In code, toClientResponse is used for every query of repair request, which including create query and modify query, and this is written by Alex. So I think it is clear that in the create and modify request, we will not need to prisma include images and even presign the image. Or maybe we don't have to use toClientResponse everywhere, I will check it later.

Reply:
Sorry, I did not realise it was used in both create and modify repair requests. I think we should be good to remove toClientResponse usage in create and modify queries and just return a status code 204